packages/dev/core/src/FlowGraph/Blocks/Execution/ControlFlow/flowGraphDebounceBlock.ts

```diff
@@ -0,0 +1,39 @@
+import type { FlowGraphContext } from "core/FlowGraph/flowGraphContext";
+import type { FlowGraphDataConnection } from "core/FlowGraph/flowGraphDataConnection";
+import { RichTypeNumber } from "core/FlowGraph/flowGraphRichTypes";
+import type { FlowGraphSignalConnection } from "core/FlowGraph/flowGraphSignalConnection";
+import { FlowGraphWithOnDoneExecutionBlock } from "core/FlowGraph/flowGraphWithOnDoneExecutionBlock";
+
+/**
+ * @experimental
+ * This block debounces the execution of a input, i.e. ensures that the input is only executed once every X times
+ */
+export class FlowGraphDebounceBlock extends FlowGraphWithOnDoneExecutionBlock {
+    public readonly count: FlowGraphDataConnection<number>;
+    public readonly reset: FlowGraphSignalConnection;
+    public readonly currentCount: FlowGraphDataConnection<number>;
+
+    constructor() {
+        super();
+        this.count = this._registerDataInput("count", RichTypeNumber);
+        this.reset = this._registerSignalInput("reset");
+        this.currentCount = this._registerDataOutput("currentCount", RichTypeNumber);
+    }
+
+    public _execute(context: FlowGraphContext, callingSignal: FlowGraphSignalConnection): void {
+        if (callingSignal === this.reset) {
+            context._setExecutionVariable(this, "debounceCount", 0);

```

Comment:
I wonder if those should be constant or inside of a construtct preventing spelling mistakes.

Reply:
This would be valid for all the nodes

Reply:
@sebavan I can do a pass during testing to move all these to constants?

Reply:
lets add a task for it to not forget